Analyze Operators + Plan Nodes

rohanaggarwal7997 opened the pull request:
This is the continuation of #704 which takes the Analyze operators through the entire pipeline. From parser , binder , optimizer and planner. The final step would be to add code generation and then analyze functionality would be good to go. Two tests are provided.

This is for #716.

17zhangw commented:
Was any conclusion ever reached about @malin1993ml's comment on #716?

rohanaggarwal7997 commented:
I thought it will be unnecessary code duplication at 3 stages of making 3 different types of nodes under the analyse. It would be convinient calling three functions directly under codegen and return the result.

malin1993ml commented:
I was thinking of only adding those nodes at the last stage of query
optimization to reuse the aggregation logic during codegen. But I’m also
fine if it’s already convenient to directly call the functions. I don’t
think you’ve touched the codegen part yet, so you can revisit this when you
do that.

On Wed, Feb 5, 2020 at 08:04 Rohan Aggarwal <notifications@github.com>
wrote:

> @rohanaggarwal7997 <https://github.com/rohanaggarwal7997> requested your
> review on: #743 <https://github.com/cmu-db/terrier/pull/743> Analyze
> front end support - Solves 716.
>
> —
> You are receiving this because your review was requested.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-db/terrier/pull/743?email_source=notifications&email_token=ABNCHTNUHOQ6GZVGZ323EKDRBK2NBA5CNFSM4KQBQQD2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOWNTX3AI#event-3009904001>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNCHTLCDJVAWMGFPYIVJKTRBK2NBANCNFSM4KQBQQDQ>
> .
>

apavlo commented on src/include/optimizer/physical_operators.h:
```diff
@@ -2051,5 +2051,71 @@ class DropView : public OperatorNode<DropView> {
+  const catalog::table_oid_t &GetTableOid() const { return table_oid_; }
+
+  /**
+   * @return trigger columns
```
This is not a trigger.

rohanaggarwal7997 replied:
ya sorry for mistake, i copy pasted lot of documentation

17zhangw commented on src/include/optimizer/logical_operators.h:
```diff
@@ -1951,4 +1951,70 @@ class LogicalDropView : public OperatorNode<LogicalDropView> {
+  const catalog::table_oid_t &GetTableOid() const { return table_oid_; }
+
+  /**
+   * @return trigger columns
```
Should probably not be "trigger" but just "columns".

17zhangw commented on src/optimizer/logical_operators.cpp:
```diff
@@ -1192,6 +1192,40 @@ bool LogicalDropView::operator==(const BaseOperatorNode &r) {
 }
 
+//===--------------------------------------------------------------------===//
+// LogicalCreateTrigger
```
Change from "LogicalCreateTrigger" to "LogicalAnalyze".

17zhangw commented on src/optimizer/physical_operators.cpp:
```diff
@@ -1220,6 +1220,40 @@ bool DropView::operator==(const BaseOperatorNode &r) {
 }
 
+//===--------------------------------------------------------------------===//
+// LogicalCreateTrigger
```
Change from "LogicalCreateTrigger" to "Analyze"

17zhangw commented on src/optimizer/query_to_operator_transformer.cpp:
```diff
@@ -584,9 +584,19 @@ void QueryToOperatorTransformer::Visit(parser::CopyStatement *op, parser::ParseR
                                        UNUSED_ATTRIBUTE parser::ParseResult *parse_result) {
   OPTIMIZER_LOG_DEBUG("Transforming AnalyzeStatement to operators ...");
+  std::vector<catalog::col_oid_t> columns;
+  auto tb_oid = accessor_->GetTableOid(op->GetAnalyzeTable()->GetTableName());
```
If there is a namespace specified, shouldn't this instead use:
`CatalogAccessor::GetTableOid(namespace_oid_t ns, std::string name)`

rohanaggarwal7997 replied:
A similar thing is done in create trigger. Does it make a difference?

rohanaggarwal7997 replied:
Now there are no more namespace id

rohanaggarwal7997 commented:
Addressed all comments

apavlo commented:
@rohanaggarwal7997 If you have addressed all the comments, mark them as 'resolved' on Github.

rohanaggarwal7997 commented:
> @rohanaggarwal7997 If you have addressed all the comments, mark them as 'resolved' on Github.

done

mbutrovich commented on src/include/optimizer/logical_operators.h:
```diff
@@ -1951,4 +1951,70 @@ class LogicalDropView : public OperatorNode<LogicalDropView> {
+  /**
+   * OID of the namespace
+   */
+  catalog::namespace_oid_t namespace_oid_;
```
Tables should be unique within a database so I don't think we need namespace oid here.

rohanaggarwal7997 replied:
Addressed

mbutrovich commented on src/include/optimizer/physical_operators.h:
```diff
@@ -2051,5 +2051,71 @@ class DropView : public OperatorNode<DropView> {
+  /**
+   * OID of the namespace
+   */
+  catalog::namespace_oid_t namespace_oid_;
```
See previous namespace oid comment.

rohanaggarwal7997 replied:
Addressed

rohanaggarwal7997 commented:
I dont know why but there were lot of changes in the optimizer for some reason of renamings. I had to rename almost all lines that I added. It works now. I also removed all namespace related things. Both the tests still pass. I think its good to merge.

apavlo commented:
@rohanaggarwal7997 Fix the linter error.

rohanaggarwal7997 commented:
> @rohanaggarwal7997 Fix the linter error.

Its actually not a linter error. I did not knew that we have to check warnings.txt after running doxygen.
